libnmstate/nm/checkpoint.py

```diff
@@ -93,7 +93,6 @@ def create(self):
         devs = []
         timeout = self._timeout
         cp_flags = (
-            CHECKPOINT_CREATE_FLAG_DESTROY_ALL |
```

Comment:
> Not destroying other checkpoints will make checkpoint creation fail if
> there is already a checkpoint for all devices. This prevents other users
> to use another instance of Nmstate to change the state while a state
> change is in progress.

Some other user of NM may have created such a checkpoint and not nmstate, so I am not clear how is this a good step here.
Previously we said something like: "If checkpoints existed before, we destroy them all and use ours."
Now we hope that there are no others which can kill us.

Reply:
Sorry, I do not follow. You believe that it is a good idea to blow up other applications that create a checkpoint by unconditionally destroying it? This sounds like a bad idea to me. IMHO the right step is to notify the user and get them to rectify the situation.

Also there is the dream of a requirement to have a global lock and now you are worried that someone else might have a smaller lock that would block Nmstate... this does not make sense or I am not understanding what you mean with `kill us`.

Reply:
Having the checkpoint created by someone else is not related to the lock, as this 3rd party could have just left the checkpoint in place and is not currently changing the system.

Per our discussion, the scenario that convinces me that this makes sense is this one:
- User calls nmstate.apply with a rollback timeout.
- User calls nmstate.apply again, before sending the confirmation, with a rollback timeout. This should fail as the user should have either confirm the changes or explicitly cause a rollback.

Note: If the use calls apply again with no timeout mentioned, no checkpoint should be created and if an existing one is already there, it should be used (in case verify or an error occurs and requires a rollback).

I suggest you document all these scenarios in the acceptance test or some other documentation so we can monitor it.

Reply:
a) I do not plan to use the timeout parameter as you mentioned but instead have an explicit commit parameter that the caller needs to set to false if they want to commit the change themself
b) Re-usinga checkpoint is something that we can discuss in the future (to me it seems wrong) and we can add it afterwards
c) I have the opinion that a checkpoint should be always created to handle verification or Nmstate runtime errors